packages/frontend-plugin-api/src/wiring/createExtensionKind.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AppNode } from '../apis';
+import { PortableSchema } from '../schema';
+import { Expand } from '../types';
+import {
+  AnyExtensionDataMap,
+  AnyExtensionInputMap,
+  ExtensionDataValues,
+  ResolvedExtensionInputs,
+  createExtension,
+} from './createExtension';
+
+/**
+ * @public
+ */
+export interface CreateExtensionKindOptions<
+  TOptions,
+  TInputs extends AnyExtensionInputMap,
+  TOutput extends AnyExtensionDataMap,
+  TConfig,
+> {
+  kind: string;
+  namespace?: string;
+  name?: string;
+  attachTo: { id: string; input: string };
+  disabled?: boolean;
+  inputs?: TInputs;
+  output: TOutput;
+  configSchema?: PortableSchema<TConfig>;
+  factory(
```

Comment:
I think most extensions will have something to put here that avoids the need to write a bit of awkward code in plugins. Even a very slim factory that just translates an option to an ouput can be pretty valuable here, because it simplifies plugin code. Remains to be seen whether there are cases where it's impossible to define a useful factory though, but I think we could leave this as a later tweak based on what we discover?

Reply:
Maybe yep.

But still, what would a card kind render by default 🤔

Reply:
This bit: https://github.com/backstage/backstage/blob/0e7340cb25e9b713579386a7e665c54be9a954fa/plugins/catalog-react/src/alpha.tsx#L94-L105